add hapi integration

rochdev opened the pull request:
This PR adds support for the `hapi` framework. Since most of the logic was already available in the `express` integration, it was extracted so it can be shared between both frameworks.

gbbr commented:
If you write documentation for each function, their arguments and how to use them from a public API standpoint I'm happy to review these for you.

rochdev commented:
Added documentation, and also refactored to make it a bit easier to read. Do note however that this is currently not a public API (only for internal use in our integrations). I will add further documentation when the API becomes public.

rochdev commented on src/plugins/util/web.js:
```diff
@@ -0,0 +1,195 @@
+    const scope = tracer.scopeManager().activate(span)
+
+    if (config.service) {
+      span.setTag(SERVICE_NAME, config.service)
```
The default service name is set by the tracer when starting any new span.

rochdev commented on src/plugins/util/web.js:
```diff
@@ -0,0 +1,195 @@
+  },
+
+  // Reactivate the request scope in case it was changed by a middleware.
+  reactivate (req) {
```
This is only used in integrations, where the scope could be changed by external factors in-between middleware.
